mmpose/datasets/pipelines/bottom_up_transform.py

```diff
@@ -242,6 +242,84 @@ def __call__(self, joints):
         return visible_kpts
 
 
+class PAFGenerator:
+    """Generate part affinity fields.
+
+    Args:
+        output_res (int): Size of feature map.
+        limb_width (int): Limb width of part affinity fields.
+        skeleton (list[list]): connections of joints.
+    """
+
+    def __init__(self, output_res, limb_width, skeleton):
+        self.output_res = output_res
+        self.limb_width = limb_width
+        self.skeleton = skeleton
+
+    def _accumulate_paf_map(self, pafs, src, dst, count):
+        """Accumulate part affinity fields between two given joints.
+
+        Args:
+            pafs (np.ndarray[2xHxW]): paf maps (2 dimensions:x axis and
+                y axis) for a certain limb connection.
+            src (np.ndarray[2,]): coordinates of the source joint.
+            dst (np.ndarray[2,]): coordinates of the destination joint.
+            count (np.ndarray[HxW]): count map that preserves the number of
+                non-zero vectors at each point
+        """
+        limb_vec = dst - src
+        norm = np.linalg.norm(limb_vec)
+        unit_limb_vec = limb_vec / norm
+
+        min_x = max(np.floor(min(src[0], dst[0]) - self.limb_width), 0)
+        max_x = min(
+            np.ceil(max(src[0], dst[0]) + self.limb_width),
+            self.output_res - 1)
+        min_y = max(np.floor(min(src[1], dst[1]) - self.limb_width), 0)
+        max_y = min(
+            np.ceil(max(src[1], dst[1]) + self.limb_width),
+            self.output_res + 1)
+
+        range_x = list(range(int(min_x), int(max_x + 1), 1))
+        range_y = list(range(int(min_y), int(max_y + 1), 1))
+        xx, yy = np.meshgrid(range_x, range_y)
+        delta_x = xx - src[0]
+        delta_y = yy - src[1]
+        dist = np.abs(delta_x * unit_limb_vec[1] - delta_y * unit_limb_vec[0])
+        mask_local = (dist < self.limb_width)
+        mask = np.zeros_like(count, dtype=bool)
+        mask[xx, yy] = mask_local
+
+        pafs[0, mask] += unit_limb_vec[0]
+        pafs[1, mask] += unit_limb_vec[1]
+        count[mask] += 1
```

Comment:
This will change the `count` passed from user. Is it intended? If so, need to add a note in docstring.

Reply:
You can use `count = count + mask`.